packages/esm-ward-app/src/types/index.ts

```diff
@@ -163,3 +164,24 @@ export interface EncounterRole extends OpenmrsResourceStrict {
   description?: string;
   retired?: boolean;
 }
+
+export interface EncounterPayload {
+  encounterDatetime: string; // date and time the encounter was created (ISO8601 Long) (REQUIRED)
+  encounterType: string; // uuid of the encounter type - initial visit, return visit etc. (REQUIRED)
+  patient: string; // the patient to whom the encounter applies
+  location: string; // the location at which the encounter occurred (REQUIRED)
+  encounterProviders: Array<{ encounterRole: string; provider: string }>; // array of providers and their role within the encounter. At least 1 provider is required
+  obs: Array<ObsPayload>; // array of observations and values for the encounter
+  form?: string; // target form uuid to be filled for the encounter
+  orders?: Array<any>; // list of orders created during the encounter
+  visit?: string; // when creating an encounter for a specific visit, this specifies the visit
```

Comment:
You can remove the comments in the interface, as most developers are/should be thorough with core structures (Encounters, Visits, etc...)

Also, a tip, if you want to specify a property, use [JSDocs](https://jsdoc.app/about-getting-started) for the same. Eg.

```
/**
 * date and time the encounter was created (ISO8601 Long) 
 */
encounterDatetime: string; 
```